frontend/screens/Register.tsx

```diff
@@ -23,62 +27,92 @@ const Register = () => {
   const navigate = useNavigate()
 
   async function handleRegister() {
-    console.log(`Register, ${username}, ${password}, ${validPassword}`)
+    console.log(`Register, ${username}, ${password}, ${firstName},${lastName},${email}`)
+    setEmailError('');
+    setPasswordError('');
     setLoading(true)
     
     // Validate the password
     if (password !== validPassword) {
-      setPasswordError('Passwords do not match')
-      setLoading(false)
-      return
-    } else {
-      setPasswordError('')
-    }
+        setPasswordError('Passwords do not match');
+        setLoading(false);
+        return;
+      }
 
     // Call api to register the user
-    try {
-      const response = await api.post("/api/user/register/", { username, password })
-      console.log(`Response: ${response.data}`)
-      navigate("/login")
-    } catch (error) {
-      console.log(error)
+    const payload = {
+        username: username,
+        password: password,
+        first_name: firstName,
+        last_name: lastName,
+        email: email,
+    };
+
+  // Call api to register the user
+  try {
+    await api.post("/api/user/register/", payload);
+    navigate("/login");
+  } catch (error) {
+    // Email error
+    if (error.response && error.response.data.email) {
+      setEmailError(error.response.data.email[0]);
```

Comment:
What is the output of this?

Reply:
I think a generic "Enter a valid email" is fine

Reply:
It captures the api response in case of the error and shows that similar to here: 
<img width="253" alt="image" src="https://github.com/richardl2003/cypherchat/assets/92397236/236892aa-0c9e-452e-b5e5-4898c6e372eb">
I can hard code it though.

Reply:
Looks perfect actually!